zerver/lib/transfer.py

```diff
@@ -83,6 +84,28 @@ def _transfer_message_files_to_s3(attachment: Attachment) -> None:
                 storage_class=settings.S3_UPLOADS_STORAGE_CLASS,
             )
             logging.info("Uploaded message file in path %s", file_path)
+        thumbnail_dir = os.path.join(settings.LOCAL_FILES_DIR, "thumbnail", attachment.path_id)
+        if os.path.isdir(thumbnail_dir):
+            thumbnails = 0
+            for thumbnail_path in glob(os.path.join(thumbnail_dir, "*")):
+                with open(thumbnail_path, "rb") as f:
+                    guessed_type = guess_type(thumbnail_path)[0]
```

Comment:
Should we be guessing the content-type, or reading that from the `ImageAttachment` table?

Reply:
The content-type stored in the ImageAttachment data is currently derived from the file's extension:
https://github.com/zulip/zulip/blob/2fdf179d42f447af5bc7b9adba8283ff6a697b26/zerver/worker/thumbnail.py#L108-L124

I was also trying to avoid having to prefetch the ImageAttachment, since we can't do that inside `_transfer_message_files_to_s3` due to the `ProcessPoolExecutor`.

Reply:
Hmm, OK, let's at least leave a comment here, that we're relying on our thumbnails always having guessable content-type, in order to avoid fetching the `ImageAttachment` inside a `ProcessPoolExecutor`.